File: app/src/main/java/com/asfoundation/wallet/router/TransactionsRouter.java
package com.asfoundation.wallet.router;

import android.content.Context;
import android.content.Intent;
import com.asfoundation.wallet.ui.TransactionsActivity;

public class TransactionsRouter {
  public void open(Context context, boolean isClearStack) {
    Intent intent = new Intent(context, TransactionsActivity.class);
    if (isClearStack) {
      intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TOP);
    }
    context.startActivity(intent);
  }

  public void navigateFromAppOpening(Context context) {
    Intent intent = TransactionsActivity.newIntent(context, false, true);
    intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TOP);
    context.startActivity(intent);
  }
}

File: app/src/main/java/com/asfoundation/wallet/router/TransactionsRouter.kt
package com.asfoundation.wallet.router

import android.content.Context
import android.content.Intent
import com.asfoundation.wallet.main.MainActivity
import javax.inject.Inject

class TransactionsRouter @Inject constructor() {

  fun navigateFromSplash(context: Context, fromSupportNotification: Boolean) {
    val intent = MainActivity.newIntent(context, fromSupportNotification)
    intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TOP
    context.startActivity(intent)
  }
}